project-koku/koku
# [Cost-905] GCP Trino

## Summary
This adds GCP processing and summarization to our Trino flow.

## Testing
1. Set envs

```
ENABLE_PARQUET_PROCESSING=True
S3_BUCKET_NAME=koku-bucket
S3_ENDPOINT=http://kokuminio:9000
S3_ACCESS_KEY=kokuminioaccess
S3_SECRET=kokuminiosecret
```

2.  make docker-up-min-presto
3. make create-test-customer
4. make load-test-customer-data start=2021-02-01 end=2021-02-15
5. Check reports APIs for each source type

6. Load real GCP source 

```
{
    "name": "GCP Test",
    "source_type": "GCP",
    "authentication": {
        "credentials": {
            "project_id": "{project}"
        }
    },
    "billing_source": {
        "data_source": {
            "dataset": "{dataset}",
            "table_id": "{table}"
        }
    }
}
```

7. Check APIs
